app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -303,6 +303,18 @@ public synchronized void remove(final int index) {
         broadcast(new RemoveEvent(index, getIndex()));
     }
 
+    /**
+     * Removes all the items from the play queue.
+     */
+    public synchronized void removeAll() {
+        while (streams.size() > 0) {
+            final int index = streams.size() - 1;
+            removeInternal(index);
+            broadcast(new RemoveEvent(index, getIndex()));
+        }
+    }
```

Comment:
Can't this be done in a way which only requires a single broadcast instead of one for each element?

Reply:
I hope it is ok. I thought this task is easier in the beginning, I don't know anything about rxJava :(